src/services/positionService.ts

```diff
@@ -697,6 +721,79 @@ export default class PositionService {
     );
   }
 
+  async buildXcallDepositParams(
+    fromToken: Token,
+    toToken: Token,
+    fromValue: string,
+    frequencyType: BigNumber,
+    frequencyValue: string,
+    destinantionChainID: number,
+    yieldFromPossible?: string,
+    yieldToPossible?: string
+  ) {
+    const destinationUSDC = this.connextService.getNativeUSDCAddress(destinantionChainID);
+
+    const { forwardCallData, totalAmmount } = await this.buildDepositParams(
+      fromToken,
+      toToken,
+      fromValue,
+      frequencyType,
+      frequencyValue,
+      yieldFromPossible,
+      yieldToPossible,
+      destinationUSDC,
+      destinantionChainID
+    );
+
+    const xTargetAddress = X_TARGET_ADDRESS[destinantionChainID]; // polygon address for mean target
+
+    const chainID = (await this.providerService.getNetwork()).chainId;
+    const DestinationToken = getWrappedProtocolToken(destinantionChainID); // can have any address in UI for destination
```

Comment:
then lets accept a new parameter for this function that is `fundWithToken` which is a `Token` and is the address of the token that we are going to fund it with.

The `Token` type already has a chainid so you can infer it from there and not have to pass a specific chain as a parameter

Reply:
So I think `from` parameter is the same, Cant we use `chainID` from `from` parameter and can do like `if(from.chainID !== to.chainID)`.